oauth2client/service_account.py

```diff
@@ -61,31 +61,49 @@ def __init__(self, service_account_id, service_account_email, private_key_id,
   def _generate_assertion(self):
     """Generate the assertion that will be used in the request."""
 
+    now = int(time.time())
     header = {
         'alg': 'RS256',
         'typ': 'JWT',
         'kid': self._private_key_id
     }
-
-    now = int(time.time())
     payload = {
         'aud': self._token_uri,
-        'scope': self._scopes,
         'iat': now,
         'exp': now + _ServiceAccountCredentials.MAX_TOKEN_LIFETIME_SECS,
         'iss': self._service_account_email
     }
-    payload.update(self._kwargs)
-
-    first_segment = _urlsafe_b64encode(_json_encode(header))
-    second_segment = _urlsafe_b64encode(_json_encode(payload))
-    assertion_input = first_segment + b'.' + second_segment
-
-    # Sign the assertion.
-    rsa_bytes = rsa.pkcs1.sign(assertion_input, self._private_key, 'SHA-256')
-    signature = base64.urlsafe_b64encode(rsa_bytes).rstrip(b'=')
-
-    return assertion_input + b'.' + signature
+    if self._scopes:
```

Comment:
The whole point of a helper is to encapsulate an action into something concrete, e.g. `_make_assertion_for_jwt` and `_use_jwt_as_assertion`. Then each of those concrete actions can be analyzed in isolation.